include/zephyr/drivers/adc.h

```diff
@@ -144,6 +144,186 @@ struct adc_channel_cfg {
 #endif /* CONFIG_ADC_CONFIGURABLE_INPUTS */
 };
 
+/**
+ * @brief Get the channel configuration from a given devicetree node
+ *
+ * This returns a static initializer for a struct adc_channel_cfg filled
+ * with data from a given devicetree node.
+ *
+ * Example devicetree fragment:
+ *
+ * @code{.dts}
+ * &adc0 {
+ *    #address-cells = <1>;
+ *    #size-cells = <0>;
+ *
+ *    channel@0 {
+ *        reg = <0>;
+ *        gain = "ADC_GAIN_1_6";
+ *        reference = "ADC_REF_INTERNAL";
+ *        acquisition-time = <ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 20)>;
+ *        differential;
+ *        input-positive = <NRF_SAADC_AIN6>;
+ *        input-negative = <NRF_SAADC_AIN7>;
+ *    };
+ * };
+ * @endcode
+ *
+ * Example usage:
+ *
+ * @code{.c}
+ * static const struct adc_channel_cfg ch0_cfg_dt =
+ *     ADC_CHANNEL_CFG_DT_GET(DT_CHILD(DT_NODELABEL(adc0), channel_0));
```

Comment:
It is an interesting idea to re-use the adc_channel_cfg structure without having to create a new adc_channel_cfg_dt. This does introduce an asymmetry with other APIs which do have foo_dt structures like gpio_dt_spec, spi_dt_spec, and i2c_dt_spec, and associated helper macros GPIO_DT_SPEC_GET, SPI_DT_SPEC_GET, I2C_DT_SPEC_GET respectively. The helper macros below do not seem to follow the pattern either.

Was this difference with other APIs intentional? If so, the commit log doesn't seem to explain why you made this decision. If not, I wonder if trying to follow the pattern with a `struct adc_channel_dt_spec` and `ADC_CHANNEL_DT_SPEC_GET` set of macros, or something similar, would feel more familiar to users of other APIs. Maybe it's not possible and I'm missing something?

Reply:
This macro is actually not something like `SPI_DT_SPEC_GET` or `I2C_DT_SPEC_GET`, it is more like `SPI_CONFIG_DT`, which creates an `spi_config` structure initializer. I see that this `_GET` suffix is confusing, so I will drop it.

I'll try to follow the "dt_spec" pattern with other macros that provide initialization for io-channels. Currently, there is only `ADC_IO_CHANNEL_DT_ARRAY_GET` (I'm not sure how to name it better) that returns an array of `adc_io_channel_dt` structures (I'll rename the structure to `adc_io_channel_dt_spec`), but I think I should also add macros that would provide initialization for a single io-channel, something like `ADC_IO_CHANNEL_DT_SPEC_GET` and `ADC_IO_CHANNEL_DT_SPEC_GET_BY_IDX`.